azkaban-common/src/test/java/azkaban/project/MockProjectLoader.java

```diff
@@ -37,30 +39,52 @@ public MockProjectLoader(File dir) {
     this.dir = dir;
   }
 
+  private ConcurrentHashMap<Integer, Project> projectsById =
+      new ConcurrentHashMap<>();
+  private ConcurrentHashMap<String, Project> projectsByName =
+      new ConcurrentHashMap<>();
+
+  private static int projectId = 0;
+
   @Override
   public List<Project> fetchAllActiveProjects() throws ProjectManagerException {
-    // TODO Auto-generated method stub
-    return null;
+    ArrayList<Project> activeProjects = new ArrayList<>();
+    for(Project project : projectsById.values()){
+      if(project.isActive()){
+        activeProjects.add(project);
+      }
+    }
+    return activeProjects;
   }
 
   @Override
   public Project fetchProjectById(int id) throws ProjectManagerException {
-    // TODO Auto-generated method stub
-    return null;
+    System.out.println("MockProjectLoader: fetch project by id " + id);
```

Comment:
The output of System.out will be redirected to a local file. This is already implemented for all other unit tests.